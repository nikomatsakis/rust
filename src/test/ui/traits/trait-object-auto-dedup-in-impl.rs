// Checks to make sure that `dyn Trait + Send` and `dyn Trait + Send + Send` are the same type.
// See issue #47010.

struct Struct;

impl Trait for Struct {}

trait Trait {}

type Send1 = Trait + Send;
type Send2 = Trait + Send + Send;

fn main () {}

impl Trait + Send {
    fn test(&self) {
    //~^ ERROR duplicate definitions with name `test`
        println!("one");
    }
}

impl Trait + Send + Send {
    fn test(&self) {
        println!("two");
    }
}
