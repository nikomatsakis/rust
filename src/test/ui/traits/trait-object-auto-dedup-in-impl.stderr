error[E0592]: duplicate definitions with name `test`
  --> $DIR/trait-object-auto-dedup-in-impl.rs:16:5
   |
LL | /     fn test(&self) {
LL | |     //~^ ERROR duplicate definitions with name `test`
LL | |         println!("one");
LL | |     }
   | |_____^ duplicate definitions for `test`
...
LL | /     fn test(&self) {
LL | |         println!("two");
LL | |     }
   | |_____- other definition for `test`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0592`.
